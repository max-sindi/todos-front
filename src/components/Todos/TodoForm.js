import React, { Component } from 'react'
import FormPartials from '../partials/Form/'
const { Input } = FormPartials

class TodoForm extends Component {

  state = {
    title: '',
    body: '',
  }

  handleSubmit = e => {
    e.preventDefault()
  }

  handleChange = e => {
    this.setState({
      [e.target.name]: e.target.value
    })
  }

  render() {
    const { fields } = this.props

    return (
      <form onSubmit={this.handleSubmit}>
        {/* mapping inputs */}
        {fields.map(field => {
          return (
            <input
              name={field.id}
              key={field.id}
              value={this.state[field.id]}
              onChange={this.handleChange}
            />
          )
        })}
      </form>
    )
  }
}

export default TodoForm
