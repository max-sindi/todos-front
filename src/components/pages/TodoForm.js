import React, { Component } from 'react'

class TodoForm extends Component {

  state = {
    title: '',
    body: '',
  }

  handleSubmit = e => {
    e.preventDefault()
  }

  handleChange = e => {
    this.setState({ [e.target.name]: e.target.value })
  }

  render() {
    const { fields } = this.props

    return (
      <form action="#" onSubmit={this.handleSubmit}>
        {fields.map(field => {
          return (
            <input value={this.state[field.id]} onChange={this.handleChange} />
          )
        })}
      </form>
    )
  }
}

export default TodoForm
