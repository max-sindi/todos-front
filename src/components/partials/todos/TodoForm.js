import React from 'react'
import { connect } from 'react-redux'
import { changeFormValue, clearTodoForm, resetEditing } from 'store/todos/actions'

const TodoForm = ({
  clearTodoForm,
  changeFormValue,
  handleSubmit,
  isEditMode,
  resetEditing,
  form: {
    body,
    isDone,
    title,
  }
}) => {
  const firstInput = React.createRef()
  const handleInputChange = ({ target: { name, value } }) => changeFormValue({
    name, value
  })

  const handleCheckboxChange = ({ target: { name, checked }}) => changeFormValue({
    name, value: checked
  })

  const callHandleSubmit = e => {
    e.preventDefault()
    handleSubmit()
      // BUG: when form called in TodoNewPage, current === null
      .then(res => firstInput.current && firstInput.current.focus())
  }

  return (
    <form onSubmit={callHandleSubmit}>
      <input
        ref={firstInput}
        value={title.value}
        name="title"
        placeholder="Enter todo's title"
        onChange={handleInputChange} />
      <input
        value={body.value}
        name="body"
        placeholder="Enter todo's body"
        onChange={handleInputChange} />
      <input
        name="isDone"
        type="checkbox"
        checked={isDone.value}
        onChange={handleCheckboxChange} />

      <button type="submit">Confirm</button>
      <button type="button" onClick={clearTodoForm}>Clear</button>
      {
        isEditMode && (
          <button type="button" onClick={resetEditing}>Reset Editing</button>
        )
      }
    </form>
  )
}

export default connect(
  store => ({
    form: store.todos.form
  }),
  { changeFormValue, clearTodoForm, resetEditing }
)(TodoForm)
