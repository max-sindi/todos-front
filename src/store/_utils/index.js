export getFormData from './getFormData'
export requestResultAction from './requestResultAction'
export handlingRequestResult from './handlingRequestResult'
export createAction from './createAction'
export createFilterValueToSend from './createFilterValueToSend'
