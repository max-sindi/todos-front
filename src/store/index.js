import {compose, applyMiddleware, createStore} from "redux"
import thunk from "redux-thunk"
import {createLogger} from "redux-logger"
import createSaga from 'redux-saga'
import rootSaga from './saga'
import combinedReducers from "./reducers"

const logger = createLogger({diff: true, collapsed: true})
const saga = createSaga()
const reduxDevtools = window.__REDUX_DEVTOOLS_EXTENSION__

const store = createStore(
  combinedReducers,
  undefined,
  compose(
    applyMiddleware( thunk, saga, logger),
    reduxDevtools ? reduxDevtools() : f => f
  )
)

saga.run(rootSaga)

export default store