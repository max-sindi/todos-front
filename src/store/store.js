import { createStore, applyMiddleware, compose } from 'redux'
import thunk from 'redux-thunk'
import reducers from './reducers'
import { createLogger } from 'redux-logger'

const logger = createLogger({
  diff: true,
  collapsed: true,
})

const middlewares = [
  thunk,
  logger,
]

const reduxDevtools = window.__REDUX_DEVTOOLS_EXTENSION__

const store = createStore(
  reducers,
  undefined,
  compose(
    applyMiddleware(...middlewares),
    reduxDevtools ? reduxDevtools() : f => f
  )
)

export default store
