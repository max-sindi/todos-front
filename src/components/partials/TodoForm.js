import React from 'react'
import { connect } from 'react-redux'
import { changeFormValue } from 'store/todos/actions'

const TodoForm = ({
  clearTodoForm,
  changeFormValue,
  handleSubmit,
  isEditMode,
  form: {
    title,
    body,
  }
}) => {
  const handleInputChange = ({ target: { name, value } }) => changeFormValue({
    name, value
  })

  const callHandleSubmit = e => {
    e.preventDefault()
    handleSubmit()
  }

  return (
    <form onSubmit={callHandleSubmit}>
      <input
        value={title.value}
        name="title"
        placeholder="Enter todo's title"
        onChange={handleInputChange}/>
      <input
        value={body.value}
        name="body"
        placeholder="Enter todo's body"
        onChange={handleInputChange}/>
      <button type="submit">Confirm</button>
      <button type="button" onClick={clearTodoForm}>Clear</button>
    </form>
  )
}

export default connect(
  store => ({
    form: store.todos.form
  }),
  { changeFormValue }
)(TodoForm)
