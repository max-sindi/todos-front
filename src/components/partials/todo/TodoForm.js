import React, {Component} from 'react';
import {Loader} from "components/partials"
import {form} from "components/partials"
import PropTypes from 'prop-types';

const {Input} = form

class TodoForm extends Component {
  state = {
    form: {
      title: '',
      body: '',
    }
  }

  componentDidUpdate = (prevProps) => {
    // push todo data to state
    if(this.props.todo !== prevProps.todo) {
      this.setState(state => ({...state, form: {...state.form, ...this.props.todo}}))
    }
  }

  static defaultProps = {
    todo: null,
  }

  static propTypes = {
    onSubmit: PropTypes.func.isRequired,
    isFetching: PropTypes.bool,
    // object with todoData, can't set null as available type
    // todo: PropTypes.oneOf([null, PropTypes.object]).isRequired,
  }

  onSubmit = (e) => {
    e.preventDefault()
    this.props.onSubmit(this.state.form)
  }

  onChangeAction = e => {
    const {name, value} = e.target
    this.setState(state => ({...state, form: {...state.form, [name]: value}}))
  }

  render() {
    const {title, body} = this.state.form
    return (
      <div>
        {this.props.isFetching ? <Loader /> : (
          <form onSubmit={this.onSubmit}>
            <Input name={"title"} value={title} onChange={this.onChangeAction} />
            <Input name={"body"} value={body} onChange={this.onChangeAction} />
            <button type={"submit"}>Update</button>
          </form>
        )}
      </div>
    );
  }
}

export default TodoForm