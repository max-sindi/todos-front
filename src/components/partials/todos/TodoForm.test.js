import React from 'react'
import TodoForm from './TodoForm'
import renderer from 'react-test-renderer'
import { Provider } from 'react-redux'
import store from 'store/store'

test('renders correctly', () => {
  const component = renderer.create(
    <Provider store={store}>
      <TodoForm />
    </Provider>
  )

  let tree = component.toJSON()
  console.log(component.toTree());

  // expect(tree).toMatchSnapshot()
})
