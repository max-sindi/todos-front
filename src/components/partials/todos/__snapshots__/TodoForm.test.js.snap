// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    name="title"
    onChange={[Function]}
    placeholder="Enter todo's title"
    value=""
  />
  <input
    name="body"
    onChange={[Function]}
    placeholder="Enter todo's body"
    value=""
  />
  <button
    type="submit"
  >
    Confirm
  </button>
  <button
    onClick={[Function]}
    type="button"
  >
    Clear
  </button>
</form>
`;
